Shared/Samples/Run valve isolation trace/RunValveIsolationTraceView.Model.swift

```diff
@@ -0,0 +1,382 @@
+        @Published var statusText: String = "Loading Utility Network…"
+        
+        /// The filter barrier identifer.
+        private static let filterBarrierIdentifier = "filter barrier"
+        
+        /// The graphic overlay to display starting location and filter barriers.
+        let parametersOverlay: GraphicsOverlay = {
+            let barrierPointSymbol = SimpleMarkerSymbol(style: .x, color: .red, size: 20)
+            let barrierUniqueValue = UniqueValue(
+                description: "Filter Barrier",
+                label: "Filter Barrier",
+                symbol: barrierPointSymbol,
+                values: [filterBarrierIdentifier]
+            )
+            let startingPointSymbol = SimpleMarkerSymbol(style: .cross, color: .green, size: 20)
+            let renderer = UniqueValueRenderer(
+                fieldNames: ["TraceLocationType"],
+                uniqueValues: [barrierUniqueValue],
+                defaultLabel: "Starting Location",
+                defaultSymbol: startingPointSymbol
+            )
+            let overlay = GraphicsOverlay()
+            overlay.renderer = renderer
+            return overlay
+        }()
+        
+        init() {
+            map.addUtilityNetwork(utilityNetwork)
+        }
+        
+        deinit {
+            ArcGISEnvironment.authenticationManager.arcGISCredentialStore.removeAll()
+        }
+        
+        /// Performs important tasks including adding credentials, loading and adding operational layers.
+        func setup() async throws {
+            try await ArcGISEnvironment.authenticationManager.arcGISCredentialStore.add(.publicSample)
+            try await loadServiceGeodatabase()
+            try await loadUtilityNetwork()
+        }
+        
+        /// Load the service geodatabase and initialize the layers.
+        @MainActor
+        private func loadServiceGeodatabase() async throws {
+            // Loads the geodatabase if it does not exist.
+            if serviceGeodatabase == nil {
+                serviceGeodatabase = ServiceGeodatabase(url: .featureServiceURL)
+                try await serviceGeodatabase.load()
+                tracingActivity = .loadingNetwork
+            }
+            
+            // The gas device layer and gas line layer are created from the service geodatabase.
+            if let gasDeviceLayerTable = serviceGeodatabase.table(withLayerID: 0),
+               let gasLineLayerTable = serviceGeodatabase.table(withLayerID: 3) {
+                let layers = [gasLineLayerTable, gasDeviceLayerTable].map(FeatureLayer.init)
+                // Add the utility network feature layers to the map for display.
+                map.addOperationalLayers(layers)
+                self.layers = layers
+            }
+        }
+        
+        /// Load the utility network.
+        @MainActor
+        private func loadUtilityNetwork() async throws {
+            try? await utilityNetwork.load()
```

Comment:
if load throws then the error is eaten and not propagated.  Then the status text is set to "Utility network loaded..." This probably should be `try await ...`